tests/utils.py

```diff
@@ -1,21 +1,25 @@
+# sourcery skip: require-return-annotation
 import functools
 import os
 
 import pytest
 
+from autogpt.config import Config
+from tests.vcr_helper import use_vcr
+
 
 def requires_api_key(env_var):
     def decorator(func):
         @functools.wraps(func)
-        def wrapper(*args, **kwargs):
-            if not os.environ.get(env_var):
-                pytest.skip(
-                    f"Environment variable '{env_var}' is not set, skipping the test."
-                )
-            else:
-                return func(*args, **kwargs)
-
-        return wrapper
+        def wrapper(request, *args, **kwargs):
+            use_api_key = request.config.getoption("--use-api-key")
+            if use_api_key:
+                return func(request, *args, **kwargs)
+            Config().set_openai_api_key("sk-dummy")
```

Comment:
Because the Config is a singleton, this mutates the api key for all future calls in the test suite after a decorated function is called.